tests/unit/amo/reducers/test_users.js

```diff
@@ -91,6 +98,56 @@ describe(__filename, () => {
 
       expect(state.byUsername).toHaveProperty('john', 12345);
     });
+
+    it('sets `resetStateOnNextChange` to `true` after a location change on the client', () => {
+      const _config = getFakeConfig({ server: false });
+
+      const state = reducer(undefined, { type: LOCATION_CHANGE }, _config);
+
+      expect(state.resetStateOnNextChange).toEqual(true);
+    });
+
+    it('does not set `resetStateOnNextChange` to `true` after a location change on the server', () => {
+      const _config = getFakeConfig({ server: true });
+
+      const state = reducer(undefined, { type: LOCATION_CHANGE }, _config);
+
+      expect(state.resetStateOnNextChange).toEqual(false);
+    });
+
+    it('resets `currentUserWasLoggedOut` after two location changes on the client', () => {
+      const _config = getFakeConfig({ server: false });
+      const { store } = dispatchSignInActions();
+      store.dispatch(logOutUser());
+
+      let state = store.getState().users;
+      expect(state.currentUserWasLoggedOut).toEqual(true);
+      expect(state.resetStateOnNextChange).toEqual(false);
+
+      // Perform two client-side location changes.
+      state = reducer(state, { type: LOCATION_CHANGE }, _config);
+      expect(state.resetStateOnNextChange).toEqual(true);
+
+      state = reducer(state, { type: LOCATION_CHANGE }, _config);
+      expect(state.currentUserWasLoggedOut).toEqual(false);
+      expect(state.resetStateOnNextChange).toEqual(false);
+    });
+
+    it('does not reset `currentUserWasLoggedOut` after only one location changes on the client', () => {
+      const _config = getFakeConfig({ server: false });
+      const { store } = dispatchSignInActions();
+      store.dispatch(logOutUser());
+
+      const state = store.getState().users;
+      expect(state.currentUserWasLoggedOut).toEqual(true);
+
+      const newState = reducer(state, { type: LOCATION_CHANGE }, _config);
+
+      expect(newState).toEqual({
```

Comment:
This is already tested and asserted in the test above for `resets currentUserWasLoggedOut after two location changes on the client`. Maybe you can just change the name of that test case and use it to test the state both after one location change and after two location changes.